analysis/02_trigger_2022_iri.md

```diff
@@ -0,0 +1,382 @@
+where values between -37.5 and 37.5 are assigned to the normal tercile.
+
+We plot the forecast raster data for the periods and leadtimes of interest.
+The red areas are the admin1's we are focussing on.
+These figures are the same as
+[the figure on the IRI Maproom](https://iridl.ldeo.columbia.edu/maproom/Global/Forecasts/NMME_Seasonal_Forecasts/Precipitation_ELR.html)
+, except for the bin boundaries.
+
+```python
+iri_dom = IriForecastDominant(constants.country_config, geo_bounding_box=geobb)
+```
+
+```python
+# iri_dom.download(clobber=True)
+# iri_dom.process(clobber=True)
+```
+
+```python
+ds_iri_dom = iri_dom.load()
+```
+
+```python
+da_iri_dom = ds_iri_dom.dominant
+```
+
+```python
+# F indicates the publication month, and L the leadtime.
+# A leadtime of 1 means a forecast published in May is forecasting JJA
+da_iri_dom_clip = da_iri_dom.rio.clip(gdf_adm1["geometry"], all_touched=True)
+```
+
+```python
+# iri website bins
+# plt_levels=[-100,-67.5,-57.5,-47.5,-42.5,-37.5,37.5,42.5,47.5,57.5,67.5,100]
+# rounded bins for easier interpretability
+plt_levels = [-100, -70, -60, -50, -45, -40, 40, 45, 50, 60, 70, 100]
+plt_colors = [
+    "#783200",
+    "#ab461e",
+    "#d18132",
+    "#e8b832",
+    "#fafa02",
+    "#ffffff",
+    "#d1f8cc",
+    "#acf8a0",
+    "#73bb6e",
+    "#3a82b3",
+    "#0e3bf4",
+]
+```
+
+```python
+def plt_single_date_leadtime(da_single):
+    g = da_single.plot(
+        levels=plt_levels,
+        colors=plt_colors,
+        cbar_kwargs={
+            "orientation": "horizontal",
+            "shrink": 0.8,
+            "aspect": 40,
+            "pad": 0.1,
+            "ticks": plt_levels,
+        },
+        figsize=(15, 7),
+    )
+
+    gdf_adm1.boundary.plot(linewidth=1, ax=g.axes, color="grey")
+    gdf_aoi.boundary.plot(linewidth=1, ax=g.axes, color="red")
+    g.axes.axis("off")
+    pub_mon = int(da_single.F.dt.month)
+    lt = int(da_single.L)
+    for_seas = utils.get_month_season_mapping()[(pub_mon + lt + 1) % 12 + 1]
+    g.axes.set_title(
+        (
+            f"Forecast published in {calendar.month_name[pub_mon]} predicting"
+            f" {for_seas} (leadtime of {lt} months)"
+        ),
+        y=1.1,
+    )
+    plt.savefig(
+        f"forecast_{str(da_mar_2022.F.dt.strftime('%b_%Y').values)}_lt_{lt}.png",
+        dpi=300,
+    )
+```
+
+```python
+da_mar_2022 = da_iri_dom_clip.sel(
+    F=cftime.Datetime360Day(2022, 3, 16, 0, 0, 0, 0), L=3
+)
+plt_single_date_leadtime(da_mar_2022)
+```
+
+```python
+da_mar_2022 = da_iri_dom_clip.sel(
+    F=cftime.Datetime360Day(2022, 7, 16, 0, 0, 0, 0), L=1
+)
+plt_single_date_leadtime(da_mar_2022)
+```
+
+### Selecting the data
+
+Before we looked at the dominant tercile.
+From here on we use a dataset that contains the probability per tercile,
+such that we can the range of values of the below average tercile,
+regardless of the values of the other terciles.
+
+We load the data and apply an approximate mask to our region of interest.
```

Comment:
how was the mask created? anything we should document in terms of parameters or decisions?

Reply:
(I know it's in the code below, but in case someone only browse the comments and/or to explain the approach and help understand the code for someone unfamiliar with the function / this type of analysis.)